packages/app/helper/useSignUp.js

```diff
@@ -0,0 +1,59 @@
+import { useReducer } from 'react'
+
+const initialState = {
+  userName: { value: '', gotTargeted: false },
+  firstName: { value: '', gotTargeted: false },
+  mail: { value: '', gotTargeted: false },
+  password: { value: '', gotTargeted: false },
+  userNameBC: '$grey80',
+  firstNameBC: '$grey80',
+  mailBC: '$grey80',
+  passwordBC: '$grey80',
+}
+
+const reducer = (state, action) => {
+  let newState
+  switch (action.type) {
+    case 'uNameChange':
+      newState = {
```

Comment:
Instead of assigning to let newState you could do an early return in your switch case statements:
```suggestion
      return {
```